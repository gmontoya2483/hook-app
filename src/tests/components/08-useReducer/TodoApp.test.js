import {mount, shallow} from "enzyme";
import TodoApp from "../../../components/08-useReducer/TodoApp";
import {demoTodos} from "../../test-data/demoTodo";
import {act} from "@testing-library/react";

describe('Pruebas en el <TodoApp />', () => {

    const wrapper = shallow(<TodoApp />);
    Storage.prototype.setItem = jest.fn(() => {});


    test('se debe de mostrar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de agregar un Todo', () => {

        const wrapper = mount(<TodoApp />);

        act( ()=>{
            wrapper.find('TodoAdd').prop('handleAddTodo')( demoTodos[0] );
            wrapper.find('TodoAdd').prop('handleAddTodo')( demoTodos[1] );
        });

        expect(wrapper.find('h1').text().trim()).toBe('TodoApp (2)')
        expect( localStorage.setItem ).toHaveBeenCalledTimes(2);

    });




    test('debe de borrar un Todo', () => {

        //Agregar un Todo
        wrapper.find('TodoAdd').prop('handleAddTodo')( demoTodos[0] );
        expect(wrapper.find('h1').text().trim()).toBe('TodoApp (1)');

        // Eliminar un Todo
        wrapper.find('TodoList').prop('handleDelete')( demoTodos[0].id );
        expect(wrapper.find('h1').text().trim()).toBe('TodoApp (0)');


    });










});
