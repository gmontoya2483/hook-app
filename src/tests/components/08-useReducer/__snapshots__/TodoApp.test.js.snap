// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <TodoApp /> se debe de mostrar correctamente 1`] = `
<div>
  <h1>
    TodoApp (
    0
    ) 
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
